File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    
    for i in range(len(arr)) :
        if len(answer) == 0 :
            answer.append(arr[i])
            continue
        
        if arr[i] == answer[-1] :
            continue
        else :
            answer.append(arr[i])
        
        
    
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {

        
        List<Integer> mylist = new ArrayList<>();
        
        int currentnum = -1;
        for(int i = 0 ; i< arr.length; i++)
        {
            if (i == 0 )
            {
                mylist.add(arr[i]);
            }
            
            else{
                if(arr[i] != currentnum)
                {
                    mylist.add(arr[i]);
                }
            }
            
              currentnum = arr[i];
        }
        
        int[]answer = new int[mylist.size()];
        
        for(int j = 0; j<mylist.size(); j++)
        {
            answer[j] = mylist.get(j);
        }

        
        
        

        return answer;
    }
}